extensions/amp-story-dev-tools/0.1/amp-story-dev-tools.js

```diff
@@ -32,21 +37,182 @@ const fontsToLoad = [
+        : null) || DevToolsTab.PREVIEW;
+
+    /** @private {!Object<string, !Element>} maps tabs to contents */
+    this.tabContents_ = {};
   }
 
   /** @override */
   buildCallback() {
     this.loadFonts_();
+    this.buildLayout_();
+    this.initializeListeners_();
+
+    this.buildTabs_();
+    this.switchTab_(this.currentTab_);
+  }
+
+  /**
+   * @private
+   */
+  buildLayout_() {
+    const container = buildContainerTemplate(this.element);
+    this.element.appendChild(container);
+    this.element.querySelector(
+      '.i-amphtml-story-dev-tools-close'
+    ).href = this.storyUrl_;
 
-    // TODO(mszylkowski): Create actual layout.
-    const message = this.element.ownerDocument.createElement('div');
-    message.textContent = 'AMP Story Dev-tools was loaded, page WIP';
-    this.element.textContent = '';
-    this.element.appendChild(message);
+    // Create tabs on top
+    const tabsContainer = container.querySelector(
+      '.i-amphtml-story-dev-tools-tabs'
+    );
+    Object.values(DevToolsTab).forEach((tabTitle) => {
+      const tabSelector = this.win.document.createElement('button');
+      tabSelector.classList.add('i-amphtml-story-dev-tools-tab-selector');
+      tabSelector.setAttribute('data-tab', tabTitle);
+      tabSelector.textContent = tabTitle;
+      tabsContainer.appendChild(tabSelector);
+    });
+  }
+
+  /**
+   * @private
+   */
+  initializeListeners_() {
+    const tabsContainer = this.element.querySelector(
+      '.i-amphtml-story-dev-tools-tabs'
+    );
+    tabsContainer.addEventListener('click', (event) => {
+      const tab = event.target.getAttribute('data-tab');
+      if (
+        Object.values(DevToolsTab).find((t) => t == tab) &&
+        this.currentTab_ != tab
+      ) {
+        this.switchTab_(tab);
+        // Update hashString with tab selected or null if tab = preview (default)
+        updateHash(
+          {
+            'tab':
+              tab == DevToolsTab.PREVIEW
+                ? null
+                : tab.toLowerCase().replace(' ', '-'),
+          },
+          this.win
+        );
+      }
+    });
+  }
+
+  /**
+   * @private
+   */
+  buildTabs_() {
+    this.tabContents_ = Object.values(DevToolsTab).reduce(
+      (tabContents, tab) => {
+        tabContents[tab] = createTabElement(this.win, this.storyUrl_, tab);
+        return tabContents;
+      },
+      {}
+    );
+  }
+
+  /**
+   * Switches the tab shown by activating the tab button and switching the contents.
+   * @param {!DevToolsTab} tab
+   */
+  switchTab_(tab) {
+    this.tabContents_[this.currentTab_].remove();
+    this.element
+      .querySelector('.i-amphtml-story-dev-tools-container')
+      .appendChild(this.tabContents_[tab]);
+    toArray(
+      this.element.querySelectorAll('.i-amphtml-story-dev-tools-tab-selector')
+    ).forEach((tabSelector) => {
+      return tabSelector.toggleAttribute(
+        'active',
+        tabSelector.getAttribute('data-tab') === tab
+      );
+    });
```

Comment:
all of this except for the querySelector should be in a `this.mutateElement(fn)`